jbmc/regression/jbmc/NondetEnumOpaqueReturn/NondetEnumOpaqueReturn.java

```diff
@@ -0,0 +1,54 @@
+public class NondetEnumOpaqueReturn {
+
+  public static void canChooseSomeConstant() {
+    Opaque o = new Opaque();
+    Color c = o.getC();
+    if (c == null)
+      return;
+    assert c != null;
+    boolean isRed = c.name().startsWith("RED") && c.name().length() == 3
```

Comment:
It's equivalent, yes, (and you wouldn't need `c.name().length() == 3`) but pointer analysis doesn't cope very well with `equals` because it's defined in `Object` and overridden almost everywhere.

Reply:
🚫  This looks like a workaround.`equals` also has to work...